### DISCUSSION TITLE
Borg Daily Backup Failed

### DISCUSSION THREAD

**Discussion by marcmatter** [nextcloud/all-in-one#2315] [Questions] [answered]
Hey!
I have been running my backups overnight for a week now and they have always been successful. but tonight the nextcloud apache container did not start up after the backup.

I did the troubleshooting below:
1. ``sudo rm /var/lib/docker/volumes/nextcloud_aio_database_dump/_data/backup-is-running``
2. restart all containers

I cloud stop the Backup but after that I started a new backup via the nextcloud aio interface and got Following Error:
```
Performing backup...
Starting the backup...
Failed to create/acquire the lock /root/.cache/borg/72fb9405c11c9c697c8c9d7835fde9bed25a806d24a04af5cda4cfbb85b029df/lock.exclusive (timeout).
Deleting the failed backup archive...
Failed to create/acquire the lock /root/.cache/borg/72fb9405c11c9c697c8c9d7835fde9bed25a806d24a04af5cda4cfbb85b029df/lock.exclusive (timeout).
Backup failed!
```

I have tried the following troubleshooting, but nothing has been working:

1. ```sudo rm /media/usb2/backup/nextcloud/borg/lock.roster```
2. ```sudo borg break-lock /media/usb2/backup/nextcloud/borg/```
3. ```sudo rm -rf /root/.cache/borg```
4. ```sudo borg check --repair /media/usb2/backup/nextcloud/borg/```
5. restart all containers

docker inspect nextcloud-aio-borgbackup:
```
...
  {
      "Type": "bind",
      "Source": "/media/usb2/backup/nextcloud",
      "Destination": "/mnt/borgbackup",
      "Mode": "rw",
      "RW": true,
      "Propagation": "rprivate"
  },
...
```

docker logs nextcloud-aio-mastercontainer 
```
++ head -1 /mnt/docker-aio-config/data/daily_backup_time
+ BACKUP_TIME=03:30
+ export BACKUP_TIME
+ export DAILY_BACKUP=1
+ DAILY_BACKUP=1
++ sed -n 2p /mnt/docker-aio-config/data/daily_backup_time
+ '[' '' '!=' automaticUpdatesAreNotEnabled ']'
+ export AUTOMATIC_UPDATES=1
+ AUTOMATIC_UPDATES=1
+ set +x
{"level":"info","ts":1681121423.9395897,"msg":"using provided configuration","config_file":"/Caddyfile","config_adapter":""}
{"level":"info","ts":1681121423.9742925,"msg":"failed to sufficiently increase receive buffer size (was: 208 kiB, wanted: 2048 kiB, got: 416 kiB). See https://github.com/quic-go/quic-go/wiki/UDP-Receive-Buffer-Size for details."}
[10-Apr-2023 10:10:24] NOTICE: fpm is running, pid 127
[10-Apr-2023 10:10:24] NOTICE: ready to handle connections
```
---------
### Host OS
Ubuntu 22.04

Nextcloud AIO version
4.7.0

Current channel
latest

-------------------

I am grateful for any further help

Greetings
Marc

**Comment by szaimen** [comment, marked as answer]
Hi, you could try to delete the borg cache which is stored in /var/lib/docker/volumes/nextcloud_aio_backup_cache/_data but I never tried that and it is possible that you would break things more by doing so.

**Reply by marcmatter** [reply]
Thank you @szaimen for your quick reply! unfortunately, the same error message still appears.
```
Performing backup...
Starting the backup...
Failed to create/acquire the lock /root/.cache/borg/72fb9405c11c9c697c8c9d7835fde9bed25a806d24a04af5cda4cfbb85b029df/lock.exclusive (timeout).
Deleting the failed backup archive...
Failed to create/acquire the lock /root/.cache/borg/72fb9405c11c9c697c8c9d7835fde9bed25a806d24a04af5cda4cfbb85b029df/lock.exclusive (timeout).
Backup failed!
```
-----
Deleting the `_data` folder causes a new error in the mastercontainer, ...
```
Code: 500
Message: Server error: `POST http://localhost/v1.41/containers/create?name=nextcloud-aio-borgbackup` resulted in a `500 Internal Server Error` response:
{&quot;message&quot;:&quot;open /var/lib/docker/volumes/nextcloud_aio_backup_cache/_data: no such file or directory&quot;}
File: /var/www/docker-aio/php/vendor/guzzlehttp/guzzle/src/Exception/RequestException.php
Line: 113
```
... so I just removed the content inside `_data`.

**Reply by szaimen** [reply]
> ... so I just removed the content inside `_data`

After doing so do you still get the same issue?

**Reply by marcmatter** [reply]
Yes

**Reply by szaimen** [reply]
So after deleting all files in this volume (including all hidden files), it still reports this error?

```
Failed to create/acquire the lock /root/.cache/borg/72fb9405c11c9c697c8c9d7835fde9bed25a806d24a04af5cda4cfbb85b029df/lock.exclusive (timeout).
```

**Reply by marcmatter** [reply]
I had to restart the mastercontainer. 🤦‍♂️  Now the backup works again!
@szaimen you are a hero! 🦸‍♂️
